### PR TITLE
Update NOTICE.txt

### PR DESCRIPTION
**Pull request by danmaas**
**Summary**
Update NOTICE.txt with latest open-source license info

**Additional Notes**
This follows the recent NOTICE.txt update to `mattermost-server`, PR here: https://github.com/mattermost/mattermost-server/pull/9433

The only difference is that this is an NPM-based project, so the dependency manifest is created by listing the "dependencies" section of all `package.json` files.

- Add new dependencies (jsc-android, moment-timezone, react-native-calendars, react-native-drawer-layout, react-navigation, rn-placeholder)
- Remove no-longer-used dependencies (react-native-drawer, react-native-fast-image)
- Update dependency with new name (TinyColor->tinycolor2)
- Update homepage and project owner info for all dependencies
- Add SPDX open-source license IDs
- Alphabetize dependency list so future update diffs will be smaller

Linebreak/whitespace changes in this PR are to match the original LICENSE files supplied with each dependency. This is a one-time change that will make future automated updates smaller.

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -7,17 +7,247 @@ NOTICES:
...
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+---
+
+## commonmark
+
+This product contains 'commonmark' by John MacFarlane.
```
Comment (enahum) [inline NOTICE.txt]: a modified version of 'commonmark'

File: NOTICE.txt [hunk 2]
```diff
@@ -7,17 +7,247 @@ NOTICES:
...
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+---
+
+lib/normalize-reference.js is a slightly modified version of
```
Comment (enahum) [inline NOTICE.txt]: I don't seem to find this one

File: NOTICE.txt [hunk 3]
```diff
@@ -7,17 +7,247 @@ NOTICES:
...
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+---
+
+lib/from-code-point.js is derived from a polyfill
```
Comment (enahum) [inline NOTICE.txt]: can't find this one either

File: NOTICE.txt [hunk 4]
```diff
@@ -7,17 +7,247 @@ NOTICES:
...
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+---
+
+bench/samples/*.md:
+
+With the exception of `bench/samples/README.md`, the samples in
```
Comment (enahum) [inline NOTICE.txt]: are these dependencies of the dependencies?

File: NOTICE.txt [hunk 5]
```diff
@@ -7,17 +7,247 @@ NOTICES:
...
+
+Released under the Creative Commons CC-BY-SA 4.0 license:
+<http://creativecommons.org/licenses/by-sa/4.0/>.
 
-## fusejs
+---
+
+## commonmark-react-renderer
+
+This product contains 'commonmark-react-renderer' by Espen Hovlandsdal.
```
Comment (enahum) [inline NOTICE.txt]: a modified version of 'commonmark-react-renderer'

File: NOTICE.txt [hunk 6]
```diff
@@ -366,199 +532,312 @@ the property of their respective owners.
...
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
 
-The licensor of the React Native software also provides the following:
+   Copyright {yyyy} {name of copyright owner}
```
Comment (enahum) [inline NOTICE.txt]: this line ??

File: NOTICE.txt [hunk 7]
```diff
@@ -799,18 +1144,20 @@ SOFTWARE.
 
 ---
 
-## react-native-notifications
+## react-native-calendars
+
+This product contains 'react-native-calendars' by Wix.com.
```
Comment (enahum) [inline NOTICE.txt]: a modified version of 'react-native-calendars'

File: NOTICE.txt [hunk 8]
```diff
@@ -897,60 +1254,55 @@ SOFTWARE.
...
-
-The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-
----
+## react-native-device-info
 
-## redux-persist
+This product contains 'react-native-device-info' by Rebecca Hughes.
```
Comment (enahum) [inline NOTICE.txt]: a modified version of 'react-native-device-info'

File: NOTICE.txt [hunk 9]
```diff
@@ -1006,84 +1359,104 @@ SOFTWARE.
...
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
 
 ---
 
-## TinyColor
+## react-native-image-gallery
 
-This product contains 'tinycolor', a small, fast library for color manipulation and conversion in JavaScript. It allows many forms of input, while providing color conversions and other color utility functions
+This product contains 'react-native-image-gallery' by Archriss.
```
Comment (enahum) [inline NOTICE.txt]: a modified version of 'react-native-image-gallery'

File: NOTICE.txt [hunk 10]
```diff
@@ -1206,75 +1585,105 @@ SOFTWARE.
...
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 ---
 
-## react-native-fast-image
+## react-native-notifications
+
+This product contains 'react-native-notifications' by Lidan Hifi.
```
Comment (enahum) [inline NOTICE.txt]: a modified version of 'react-native-notifications'

File: NOTICE.txt [hunk 11]
```diff
@@ -1527,18 +2074,20 @@ SOFTWARE.
 
 ---
 
-## react-native-safe-area
+## react-native-youtube
+
+This product contains 'react-native-youtube' by Param Aggarwal.
```
Comment (enahum) [inline NOTICE.txt]: a modified version of 'react-native-youtube'

File: NOTICE.txt [hunk 12]
```diff
@@ -2092,61 +2394,20 @@ SOFTWARE.
...
-distributed under the License is distributed on an "AS IS" BASIS, 
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
-See the License for the specific language governing permissions and 
-limitations under the License.
-
----
+## rn-fetch-blob
 
-## react-native-permissions
+This product contains 'rn-fetch-blob' by Joltup.
```
Comment (enahum) [inline NOTICE.txt]: a modified version of 'rn-fetch-blob'

File: NOTICE.txt [hunk 13]
```diff
@@ -2168,53 +2429,83 @@ SOFTWARE.
 
 ---
 
-## react-native-tableview
+## rn-placeholder
+
+This product contains 'rn-placeholder' by Marvin FRACHET.
```
Comment (enahum) [inline NOTICE.txt]: a modified version of 'rn-placeholder'

File: NOTICE.txt (hunk 1 above)
Comment (danmaas) [reply NOTICE.txt]: I'm happy to update the automation to distinguish "a modified version of..." from the unmodified upstream, where possible.

Would it be sufficient to look at the `package.json` version constraint, and if we find a version like `"github:xxxx"`, where xxxx is not the official upstream repo, then we say it's "modified"?

File: NOTICE.txt (hunk 2 above)
Comment (danmaas) [reply NOTICE.txt]: This is part of the commonmark.js LICENSE file which we include verbatim:
https://github.com/commonmark/commonmark.js/blob/master/LICENSE

It's a little awkward that upstream LICENSEs can inject confusing `---` markers. Not sure what to do about this.

File: NOTICE.txt (hunk 3 above)
Comment (danmaas) [reply NOTICE.txt]: same as above, this is just part of the commonmark.js LICENSE file

File: NOTICE.txt (hunk 4 above)
Comment (danmaas) [reply NOTICE.txt]: same as above, this is just part of the commonmark.js LICENSE file

File: NOTICE.txt (hunk 6 above)
Comment (jasonblais) [reply NOTICE.txt]: Submitted a fix: https://github.com/mattermost/mattermost-redux/pull/647

File: NOTICE.txt (hunk 1 above)
Comment (danmaas) [reply NOTICE.txt]: done, the PR is now updated with "a modified version of..." as discussed above.

File: NOTICE.txt (hunk 2 above)
Comment (hmhealey) [reply NOTICE.txt]: I'm not sure what the typical process for including upstream licenses would be. It's a bit awkward in cases, like this, but I don't think it should cause any problems outside of being a bit confusing.

**Pull request merged**